MediaBrowser.Model/Dto/MediaSourceInfo.cs

```diff
@@ -70,6 +70,8 @@ public MediaSourceInfo()
 
         public bool IsInfiniteStream { get; set; }
 
+        public bool RequiresTsContainerTranscoding { get; set; }
```

Comment:
But I don't really get the use case? We only have fmp4 and mpegts for hls container currently so it is either or.

Reply:
For me, `RequiresTsContainerTranscoding` is too specific. The container list is more generic and similar to a device profile.

Reply:
It is *that* specific because it has to be ts to workaround the compatibility issues, which is the whole reason this field is added. Or name it in other way, it would be `shouldUseMoreCompatibleTranscodingProfile`. This field's whole intent is not to override the global container preference but to enforce compatibility, and it does so by discarding all fmp4 profiles. The other way around will not work good because the clients disabled fmp4 globally usually have very specific reason and those clients usually supports fmp4 poorly, which means an override like this will likely cause more problems.

Reply:
> Or name it in other way, it would be `shouldUseMoreCompatibleTranscodingProfile`.

Now I get it. This actually sounds better if we are going to limit not only the container, but also other parameters.
I almost like `UseMostCompatibleTranscodingProfile`.

Technically, we can ask to transcode to MP4 or MKV (for downloading/recording purposes). Wouldn't one fixed container severely limit the capabilities?

Reply:
> Technically, we can ask to transcode to MP4 or MKV (for downloading/recording purposes). Wouldn't one fixed container severely limit the capabilities?

Currently this is used for LiveTV sources and unfortunately those problematic sources will have problem even transcode directly to MP4 or MKV and the audio will still be out of sync if no other post processing is applied. 

For recoding, currently all our recordings are generated as mpegts streams and some of them will also have compatibility issues but the compatibility profile is not applied yet due to the way we handles recordings(no easy way to check if the recording video is come from a specific TV Tuner). 

The capabilities will be limited due to the limitation of mpegts but it is harder to fix TV streams that does not work well with ffmpeg.

Reply:
> Currently this is used for LiveTV sources and unfortunately those problematic sources will have problem even transcode directly to MP4 or MKV and the audio will still be out of sync if no other post processing is applied.

I still believe that a device profile, but for the source (more specifically, additional container/codec conditions), will help fine tune the transcoding (workaround issues). And yes, we can get unworkable (and unavoidable) combination of device profile and source profile. Then the user should use a different device or a different source. But this can be implemented later. The current implementation is like the "rule of thumb".

> For recoding, currently all our recordings are generated as mpegts streams and some of them will also have compatibility issues but the compatibility profile is not applied yet due to the way we handles recordings(no easy way to check if the recording video is come from a specific TV Tuner).

We can't prevent someone from writing a LiveTV "download" feature in their client. But since you say there could be problems, let's forget about it.

> The capabilities will be limited due to the limitation of mpegts but it is harder to fix TV streams that does not work well with ffmpeg.

By "capabilities" I meant "remuxing to MP4/MKV". As above, let's forget about it.